src/main/java/lambdastreams/task2/Book.java

```diff
@@ -0,0 +1,111 @@
+package lambdastreams.task2;
+
+import org.apache.commons.lang3.RandomStringUtils;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+
+public class Book {
+
+    private final String name;
+    private final Integer price;
+
+    public Book(String name, Integer price) {
+        this.name = name;
+        this.price = price;
+    }
+
+    public String getName() {
+        return name;
+    }
+
+    public Integer getPrice() {
+        return price;
+    }
+
+    public static List<Book> createListOfBooks(Integer booksQuantity) {
+        List<Book> listOfBooks = new ArrayList<>();
+        for (int i = 0; i < booksQuantity; i++) {
+            String randomBookName = RandomStringUtils.randomAlphabetic(3, 10);
+            Integer randomBookPrice = new Random().nextInt(100);
+            Book book = new Book(randomBookName, randomBookPrice);
+            listOfBooks.add(book);
+        }
+        return listOfBooks;
+    }
+
+    public static List<String> createListOfNamesOfAllBooks(List<Book> listOfBooks) {
+        return listOfBooks.stream()
+                .map(Book::getName)
+                .collect(Collectors.toList());
+    }
+
+    public static int totalPriceOfAllBooks(List<Book> listOfBooks) {
+        return listOfBooks.stream()
+                .map(Book::getPrice)
+                .reduce(0, Integer::sum);
+    }
+
+    public static List<String> createListOfBooksContainsA(List<Book> listOfBooks) {
+        return listOfBooks.stream()
+                .map(Book::getName)
+                .filter(name -> name.contains("A"))
+                .collect(Collectors.toList());
+    }
+
+    public static Book selectBookWithHighestPrice(List<Book> listOfBooks) {
+        return listOfBooks.stream()
+                .max(Comparator.comparing(Book::getPrice))
+                .orElseThrow(NoSuchElementException::new);
+    }
+
+    public static int totalOfBookNamesWith5Character(List<Book> listOfBooks) {
+        return (int) listOfBooks.stream()
+                .map(Book::getName)
+                .mapToInt(String::length)
```

Comment:
this works correctly, however, just a small hint - you can remove this mapToInt() call and use name.length() on the next line for filter